### ISSUE TITLE
Modal Aria Label applied to the wrong element

### ISSUE THREAD

**Issue by NakedJavaScript** [react-bootstrap/react-bootstrap#5084]
## Describe the bug

The modal component produces a document element inside a dialog element and the aria label gets applied to the former, rather than to the later. 

## To Reproduce

1. Go to https://react-bootstrap.netlify.com/components/modal/#modal-vertically-centered
2. Click on 'Launch vertically centered modal'
3. Look at the pure HTML and you'll that the label is applied to the wrong element

## Expected behavior

See this page for an explanation for where the aria label should be set:
https://amp.levelaccess.net/public/standards/view_best_practice.php?violation_id=1893

## Environment (please complete the following information)

- Operating System: macOS
- Browser, Version: Chrome 80, Safari 13
- React-Bootstrap Version: the one that the docs used lol

**Comment by taion** [comment]
We'd be happy to take a PR to address this.

**Comment by kosmiq** [comment]
I started looking into this...

@NakedJavaScript would this be the correct markup? I haven't had the time to really dig in to the specification, so any pointers would be appreciated.

![Screenshot 2020-04-06 at 10 03 06](https://user-images.githubusercontent.com/322759/78536285-e28bf900-77ed-11ea-9a97-91be4512e471.png)

I'm a bit unsure about how to set the role tags, I haven't fiddled with them. I worked only on the "aria-modal" attribute for now.

**Comment by tim-fi** [comment]
Hi @kosmiq,

I'm a colleague of @NakedJavaScript and was pair-programming with hime when we stumbled upon this. The problem ist that the HTML that gets produce when using the `Modal`-component looks like this for use and in the docs that he referenced:
<img width="843" alt="Screen Shot 2020-04-07 at 13 48 10" src="https://user-images.githubusercontent.com/30415153/78665752-949ef000-78d6-11ea-8aa0-5422ec2ada51.png">
The problem is that the `aria-labelledby="..."` attribute gets set on the `div` with `role="document"` and that it should be set on the `div` with `role="dialog"`. At least that is our current understanding of the ARIA standard and the tools we used to check for issues seem to all support that.

PS: The [code in the TWBS docs](https://getbootstrap.com/docs/4.4/components/modal/#live-demo) actually also does this:
<img width="908" alt="Screen Shot 2020-04-07 at 13 55 36" src="https://user-images.githubusercontent.com/30415153/78666327-7b4a7380-78d7-11ea-8bd3-859ffc69ef34.png">
